packages/menu-button/src/index.js

```diff
@@ -369,6 +271,120 @@ MenuLink.propTypes = {
   _ref: func
 };
 
+////////////////////////////////////////////////////////////////////////
+
+let MenuList = props => (
+  <Consumer>
+    {({ refs, state, setState }) =>
+      state.isOpen && (
+        <Portal>
+          <WindowSize>
+            {() => (
+              <Rect>
+                {({ rect: menuRect, ref: menuRef }) => (
+                  <div
+                    data-reach-menu
+                    ref={menuRef}
+                    style={getStyles(state.buttonRect, menuRect)}
+                  >
+                    <MenuListImpl
+                      {...props}
+                      setState={setState}
+                      state={state}
+                      refs={refs}
+                    />
+                  </div>
+                )}
+              </Rect>
+            )}
+          </WindowSize>
+        </Portal>
+      )
+    }
+  </Consumer>
+);
+
+MenuList.propTypes = {
+  children: node
+};
+
+let focusableChildrenTypes = [MenuItem, MenuLink];
+
+let isFocusableChildType = child => focusableChildrenTypes.includes(child.type);
+let getFocusableMenuChildren = children => {
+  return children.filter(child => isFocusableChildType(child));
+};
+
+let MenuListImpl = React.forwardRef(
+  ({ refs, state, setState, children, onKeyDown, onBlur, ...rest }, ref) => {
+    let focusableChildren = getFocusableMenuChildren(children);
+    return (
+      <div
+        data-reach-menu-list
+        {...rest}
+        role="menu"
+        aria-labelledby={state.buttonId}
+        tabIndex="-1"
+        ref={node => {
+          refs.menu = node;
+          ref && ref(node);
+        }}
+        onBlur={event => {
+          if (
+            !state.closingWithClick &&
+            !refs.menu.contains(event.relatedTarget)
+          ) {
+            setState(close);
+          }
+        }}
+        onKeyDown={wrapEvent(onKeyDown, event => {
+          if (event.key === "Escape") {
+            setState(close);
+          } else if (event.key === "ArrowDown") {
+            event.preventDefault(); // prevent window scroll
+            let nextIndex = state.selectionIndex + 1;
+            if (nextIndex !== focusableChildren.length) {
```

Comment:
Notable change `React.Children.count(children) -> focusableChildren.length` is lost in the noise here.

I had to move MenuList below MenuItem/MenuLink in order to define and subsequently use `focusableChildrenTypes`

Reply:
Cool, this is exactly what I had in mind. Going to pull this code down and mess around w/ it a bit.